lib/db/open.go

```diff
@@ -0,0 +1,51 @@
+package db
+
+import (
+	"context"
+	"database/sql"
+	"math/rand"
+	"time"
+
+	_ "github.com/lib/pq"
+)
+
+type dbOpts struct {
+	maxConnectionLifetime       time.Duration
+	maxConnectionLifetimeJitter time.Duration
+}
+
+type PGOpenOption func(*dbOpts)
+
+func WithMaxConnectionLifetime(d time.Duration) PGOpenOption {
+	return func(opts *dbOpts) {
+		opts.maxConnectionLifetime = d
+	}
+}
+
+func WithMaxConnectionLifetimeJitter(d time.Duration) PGOpenOption {
+	return func(opts *dbOpts) {
+		opts.maxConnectionLifetimeJitter = d
+	}
+}
+
+func PGOpen(dataSourceName string) (*sql.DB, error) {
+	return PGOpenContext(context.Background(), dataSourceName)
+}
+
+func PGOpenContext(ctx context.Context, dataSourceName string) (*sql.DB, error) {
+	opts := dbOpts{
+		maxConnectionLifetime:       10 * time.Minute,
+		maxConnectionLifetimeJitter: 10 * time.Minute,
+	}
+	db, err := sql.Open("postgres", dataSourceName)
+	if err != nil {
+		return nil, err
+	}
+	db.SetConnMaxLifetime(opts.maxConnectionLifetime + jitter(opts.maxConnectionLifetimeJitter))
```

Comment:
`max` is a bit of a misnomer since we are _adding_ the jitter.  But as long as we document it, it's fine.  What is the reasoning behind wanting a jitter here?

Reply:
try to spread out when things reconnect a little bit. Maybe it doesn't matter